### ISSUE TITLE
Video progress resets when switching to popup/background and viceversa

### ISSUE THREAD

**Issue by Dangelman** [TeamNewPipe/NewPipe#7427]
# Workaround: https://github.com/TeamNewPipe/NewPipe/issues/7427#issuecomment-1014762780

### Checklist

- [x] I am using the latest version - 0.21.13 
- [x] I checked, but didn't find any duplicates (open OR closed) of this issue in the repo. 
- [x] I have read the contribution guidelines given at https://github.com/TeamNewPipe/NewPipe/blob/HEAD/.github/CONTRIBUTING.md.
- [x] This issue contains only one bug. I will open one issue for every bug report I want to file.

### Steps to reproduce the bug
1. Play a video
2. Watch if for some minutes in portrait mode
3. Switch to popup or background

### Actual behavior

The video progress will reset to either the start or the video or the last timestamp you paused it. The opposite is also true (especially switching from background to main player). I have auto-rotate disabled, if it can help.

### Expected behavior

The video should progress from the same point (or 1-2 seconds before).

### Screenshots/Screen recordings

https://github.com/TeamNewPipe/NewPipe/issues/7427#issuecomment-97482348
(change resolution if video freezes)

### Device info

 - Android version/Custom ROM version: 11
 - Device model: LM-V500EM (LG V50 ThinQ 5G)

**Comment by mhmdanas** [comment]
Sounds like a duplicate of #7303 to me, but I'm not exactly sure what the reproduction steps in your issue are since you didn't fill out that section, so please do that. :)

**Comment by Dangelman** [comment]
> Sounds like a duplicate of #7303 to me, but I'm not exactly sure what the reproduction steps in your issue are since you didn't fill out that section, so please do that. :)
mb, I forgot to delete the comment brackets.

It should be the same as #7303, tho the author did not mention the problem happening to popup or the possibility for the video to resuming from the last pause timestamp instead of restarting.

**Comment by opusforlife2** [comment]
@Dangelman Could you provide the exact steps? As I've written in the issue, it takes 2 switches for the bug to happen for me. Your steps seem to contain only one switch. Something is missing here.

**Comment by Dangelman** [comment]
@opusforlife2 It's as I described.
1. Open NewPipe
2. Select a new video
3. Play the video in portrait mode for 30s
4. Switch to either popup/background

The video restarts instead of continuing, or jump back to the last timestamp you paused previously.

**Comment by Dangelman** [comment]
https://streamable.com/dsae9q
Here's a video. As you can see, the progress is all over the place during switches.
Sorry for buffer and fullscreen, I guess the video works for issue #7404 and #7252 too.

**Comment by opusforlife2** [comment]
> The video restarts instead of continuing

This doesn't happen for me after step 4. Could you try it from a different phone?

**Comment by litetex** [comment]
Code analysis:
I think there is a problem inside https://github.com/TeamNewPipe/NewPipe/blob/d8236bbedd4713f777e90d249850b31c1f71cf3f/app/src/main/java/org/schabi/newpipe/player/Player.java#L590 which also causes #7303

When looking at this big ``if`` block https://github.com/TeamNewPipe/NewPipe/blob/d8236bbedd4713f777e90d249850b31c1f71cf3f/app/src/main/java/org/schabi/newpipe/player/Player.java#L637-L713 it seems that ``2. User changed a player from, for example. main to popup, or from audio to main, etc`` is not working. When I press "Play in background", ``1. User pressed on a timestamp link and the same video should be rewound to the timestamp`` is executed instead.
The key code-line seems to be https://github.com/TeamNewPipe/NewPipe/blob/d8236bbedd4713f777e90d249850b31c1f71cf3f/app/src/main/java/org/schabi/newpipe/player/Player.java#L650 because the recovery position is set immediately afterwards in https://github.com/TeamNewPipe/NewPipe/blob/d8236bbedd4713f777e90d249850b31c1f71cf3f/app/src/main/java/org/schabi/newpipe/player/Player.java#L718

Proposed solutions:
* Quick/Dirty one: clear recovery/playback position when changing player-type
* Better one: I think this if-block should be reworked with event-types or something similar. So that we can clearly disguise what should be done.

**Comment by rancidfrog** [comment]
For me playback resumes on pause when switching to background and vice versa.
I believe it triggers play.
1. Pause playback on pop-up viewer
2. Switch to background
3. Video is resumed automatically, without clicking play

**Comment by litetex** [comment]
@rancidfrog 
How is this related to this issue?

**Comment by rancidfrog** [comment]
@litetex 
Maybe, because switching triggers some play function.
It is related to the menu button.
Clicking the menu button points to code 
Code triggers play
Play might cause video to restart or paused video to play

HmMm, maybe

**Comment by litetex** [comment]
@rancidfrog 
I still don't get it. What has that to do with this issue or how is this additional information?
I don't know what you mean exactly but [I did a deep code analysis a few comments above](https://github.com/TeamNewPipe/NewPipe/issues/7427#issuecomment-974830456).
I'm hiding the above comments as they don't help fixing the issue.

**Comment by akberkhan-time** [comment]
`
 switch (v.getId()) {
            case R.id.detail_controls_background:
                Log.i(TAG, "onClick: PlayInBackgrount amlaml_bb1     case R.id.detail_controls_background: ");

                openBackgroundPlayer(false);
                break;
            case R.id.detail_controls_popup:
                player.saveStreamProgressState();

                openPopupPlayer(false);
                break;
`

**Comment by akberkhan-time** [comment]
after following the  [lead from litetex](https://github.com/TeamNewPipe/NewPipe/issues/7427#issuecomment-974830456) , I found the issue may caused by not properly updating the streamProgressState so i did update  it before change stream to popup or background player  and it worked .
https://github.com/TeamNewPipe/NewPipe/blob/d8236bbedd4713f777e90d249850b31c1f71cf3f/app/src/main/java/org/schabi/newpipe/fragments/detail/VideoDetailFragment.java#L439-L445

to 

```
   switch (v.getId()) {
            case R.id.detail_controls_background:
                player.saveStreamProgressState();
                openBackgroundPlayer(false);
                break;
            case R.id.detail_controls_popup:
                player.saveStreamProgressState();
                openPopupPlayer(false);
                break;
```
i'm new to this repo , would love to hear further from u guys .

**Comment by akberkhan-time** [comment]
after following the  [lead from litetex](https://github.com/TeamNewPipe/NewPipe/issues/7427#issuecomment-974830456) , I found the issue may caused by not properly updating the streamProgressState so i did update  it before change stream to popup or background player  and it worked .
https://github.com/TeamNewPipe/NewPipe/blob/d8236bbedd4713f777e90d249850b31c1f71cf3f/app/src/main/java/org/schabi/newpipe/fragments/detail/VideoDetailFragment.java#L439-L445

to 

```java
   switch (v.getId()) {
            case R.id.detail_controls_background:
                player.saveStreamProgressState();
                openBackgroundPlayer(false);
                break;
            case R.id.detail_controls_popup:
                player.saveStreamProgressState();
                openPopupPlayer(false);
                break;
```
i'm new to this repo , would love to hear further from u guys .

**Comment by tsiflimagas** [comment]
@akberkhan-time this seems to be working fine indeed, nice job! Note it should also be done for `case R.id.overlay_thumbnail:` to properly update stream progress when coming back from another player. This solution seems to be not working when there's an enqueued stream though. I'm happy this starts to be coming to a solution nevertheless, as it is a rather annoying bug.

**Comment by protodrew** [comment]
just popping in to say I have had this issue as well. Using a S20 Ultra on the latest version

**Comment by drewsat** [comment]
Downgraded to v0.21.10 worked for me

**Comment by MD77MD** [comment]
any news about a solution for this bug

**Comment by PacoBell** [comment]
Yeah, this change in v0.21.15 didn't fix the issue for me in Android 12:

"Removed MediaParser support to fix failing playback resume after buffering on Android 11+."

Please implement akberkhan-time's streamProgressState findings into a new release. I often switch between full screen, pop-up, and background modes and this bug has made the app nearly unusable for me. Until then, I'm reverting to v0.21.10 like drewsat suggested.

**Comment by opusforlife2** [comment]
> Yeah, this change in v0.21.15 didn't fix the issue for me in Android 12:

There is no way it could have. It's an unrelated PR. If you open it, you'll see the linked issue which that PR addresses.

**Comment by phyxfire** [comment]
Just wanted to share I also have had this issue. It's the worst to be watching a really long video, and then randomly have it snap back to the beginning again because you turned off the screen.

**Comment by TheShanMan** [comment]
I have this issue. Workaround: pause the video before switching how it's being played. Always solves it for me.

**Comment by opusforlife2** [comment]
@TheShanMan Awesome. I'm linking this up top.

**Comment by litetex** [comment]
I tried to quickly fix the underlying problem - where the code tries to discover what happens based on parameter that don't describe what actually happens - but the code is a spaghetti monster <sup>of doom</sup>:

<details><summary>Occasions where the player is changed or started</summary>

Summarized call hierarchy:

![grafik](https://user-images.githubusercontent.com/40789489/149827742-0d244b1e-6448-4da5-a76f-0dad3c2f6ca5.png)

or as plain text:
```
NavigationHelper.getPlayerIntent(Context, Class<T>, PlayQueue, boolean, boolean)  (org.schabi.newpipe.util)
    VideoDetailFragment.openMainPlayer()  (org.schabi.newpipe.fragments.detail)
        VideoDetailFragment.openVideoPlayer(boolean)  (org.schabi.newpipe.fragments.detail)
            VideoDetailFragment.openVideoPlayerAutoFullscreen()  (org.schabi.newpipe.fragments.detail)
                VideoDetailFragment.onServiceConnected(Player, MainPlayer, boolean)  (org.schabi.newpipe.fragments.detail)
                    PlayerHolder.setListener(PlayerServiceExtendedEventListener)  (org.schabi.newpipe.player.helper)
                    PlayerServiceConnection in PlayerHolder.onServiceConnected(ComponentName, IBinder)  (org.schabi.newpipe.player.helper)
                VideoDetailFragment.runWorker(boolean, boolean)  (org.schabi.newpipe.fragments.detail)
                    VideoDetailFragment.startLoading(boolean)  (org.schabi.newpipe.fragments.detail)
                    VideoDetailFragment.startLoading(boolean, boolean)  (org.schabi.newpipe.fragments.detail)
                VideoDetailFragment.onClick(View)  (org.schabi.newpipe.fragments.detail)
                    Anonymous in ActionBarDrawerToggle() in ActionBarDrawerToggle.onClick(View)  (androidx.appcompat.app)
                    ClickInterceptSpinner.performClick()  (com.android.printspooler.widget)
                    SearchView.onSearchClicked()  (android.widget)
                    QSFooterImpl.performAccessibilityAction(int, Bundle)  (com.android.systemui.qs)
                    Anonymous in onCreateView() in MaterialDatePicker.onClick(View)  (com.google.android.material.datepicker)
                    SearchView.onSearchClicked()  (androidx.appcompat.widget)
                    Anonymous in onCreateView() in MaterialTimePicker.onClick(View)  (com.google.android.material.timepicker)
                    TextActionModeCallback in Editor.onAssistMenuItemClicked(MenuItem)  (android.widget)
                    AccessibilityTargetHelper.createEnableDialogContentView(Context, AccessibilityServiceTarget, OnClickListener, OnClickListener)(2 usages)  (com.android.internal.accessibility.dialog)
                    KeyButtonView.onTouchEvent(MotionEvent)  (com.android.systemui.statusbar.policy)
                    Anonymous in mOnClickListener in FragmentBreadCrumbs.onClick(View)  (android.app)
                    View.callOnClick()  (android.view)
                    Anonymous in setAction() in Snackbar.onClick(View)  (com.google.android.material.snackbar)
                    RequireScrollMixinTest.testCreateOnClickListener()(2 usages)  (com.android.setupwizardlib.template)
                    TunerZenModePanel.onClick(View)  (com.android.systemui.tuner)
                    Chip.performCloseIconClick()  (com.google.android.material.chip)
                    View.performClick()  (android.view)
                    Anonymous in createOnClickListener() in RequireScrollMixin.onClick(View)  (com.android.setupwizardlib.template)
                    PassThroughClickListener in AutoCompleteTextView.onClick(View)  (android.widget)
                    PhotoView.onSingleTapConfirmed(MotionEvent)  (com.android.ex.photo.views)
                    Anonymous in onCreateView() in MaterialTimePicker.onClick(View)  (com.google.android.material.timepicker)
                    DelayedOnClickListener in SmartReplyView.onClick(View)  (com.android.systemui.statusbar.policy)
            NavigationHelper.openVideoDetailFragment(Context, FragmentManager, int, String, String, PlayQueue, boolean)  (org.schabi.newpipe.util)
                NavigationHelper.playOnMainPlayer(AppCompatActivity, PlayQueue)  (org.schabi.newpipe.util)
                    LocalPlaylistFragment.handleResult(List<PlaylistStreamEntry>)  (org.schabi.newpipe.local.playlist)
                    StatisticsPlaylistFragment.handleResult(List<StreamStatisticsEntry>)  (org.schabi.newpipe.local.history)
                    ChannelFragment.handleResult(ChannelInfo)  (org.schabi.newpipe.fragments.list.channel)
                    PlaylistFragment.handleResult(PlaylistInfo)  (org.schabi.newpipe.fragments.list.playlist)
                MainActivity.handleIntent(Intent)  (org.schabi.newpipe)
                    MainActivity.initFragments()  (org.schabi.newpipe)
                    MainActivity.onNewIntent(Intent)  (org.schabi.newpipe)
                Anonymous in initListeners() in LocalPlaylistFragment.selected(LocalItem)  (org.schabi.newpipe.local.playlist)
                    BaseFragment.onViewCreated(View, Bundle)  (org.schabi.newpipe)
                    FeedFragment.initListeners()  (org.schabi.newpipe.local.feed)
                VideoDetailFragment.onActivityResult(int, int, Intent)  (org.schabi.newpipe.fragments.detail)
                    Anonymous in attachController() in FragmentManager.onActivityResult(ActivityResult)  (androidx.fragment.app)
                    Anonymous in attachController() in FragmentManager.onActivityResult(ActivityResult)  (androidx.fragment.app)
                Anonymous in initListeners() in StatisticsPlaylistFragment.selected(LocalItem)  (org.schabi.newpipe.local.history)
                    BaseFragment.onViewCreated(View, Bundle)  (org.schabi.newpipe)
                    FeedFragment.initListeners()  (org.schabi.newpipe.local.feed)
                Anonymous in FeedFragment.kt.onItemClick(Item, View)  (org.schabi.newpipe.local.feed)
                BaseListFragment.onStreamSelected(StreamInfoItem)  (org.schabi.newpipe.fragments.list)
                    Anonymous in initListeners() in BaseListFragment.selected(StreamInfoItem)  (org.schabi.newpipe.fragments.list)
            VideoDetailFragment.onClick(View)  (org.schabi.newpipe.fragments.detail)
                Anonymous in ActionBarDrawerToggle() in ActionBarDrawerToggle.onClick(View)  (androidx.appcompat.app)
                ClickInterceptSpinner.performClick()  (com.android.printspooler.widget)
                SearchView.onSearchClicked()  (android.widget)
                QSFooterImpl.performAccessibilityAction(int, Bundle)  (com.android.systemui.qs)
                Anonymous in onCreateView() in MaterialDatePicker.onClick(View)  (com.google.android.material.datepicker)
                SearchView.onSearchClicked()  (androidx.appcompat.widget)
                TextActionModeCallback in Editor.onAssistMenuItemClicked(MenuItem)  (android.widget)
                Anonymous in onCreateView() in MaterialTimePicker.onClick(View)  (com.google.android.material.timepicker)
                AccessibilityTargetHelper.createEnableDialogContentView(Context, AccessibilityServiceTarget, OnClickListener, OnClickListener)(2 usages)  (com.android.internal.accessibility.dialog)
                KeyButtonView.onTouchEvent(MotionEvent)  (com.android.systemui.statusbar.policy)
                Anonymous in mOnClickListener in FragmentBreadCrumbs.onClick(View)  (android.app)
                View.callOnClick()  (android.view)
                Anonymous in setAction() in Snackbar.onClick(View)  (com.google.android.material.snackbar)
                RequireScrollMixinTest.testCreateOnClickListener()(2 usages)  (com.android.setupwizardlib.template)
                TunerZenModePanel.onClick(View)  (com.android.systemui.tuner)
                Chip.performCloseIconClick()  (com.google.android.material.chip)
                PassThroughClickListener in AutoCompleteTextView.onClick(View)  (android.widget)
                View.performClick()  (android.view)
                Anonymous in createOnClickListener() in RequireScrollMixin.onClick(View)  (com.android.setupwizardlib.template)
                PhotoView.onSingleTapConfirmed(MotionEvent)  (com.android.ex.photo.views)
                Anonymous in onCreateView() in MaterialTimePicker.onClick(View)  (com.google.android.material.timepicker)
                DelayedOnClickListener in SmartReplyView.onClick(View)  (com.android.systemui.statusbar.policy)
NavigationHelper.getPlayerEnqueueIntent(Context, Class<T>, PlayQueue)  (org.schabi.newpipe.util)
    NavigationHelper.enqueueOnPlayer(Context, PlayQueue, PlayerType)  (org.schabi.newpipe.util)
        LocalPlaylistFragment.handleResult(List<PlaylistStreamEntry>)(2 usages)  (org.schabi.newpipe.local.playlist)
            SearchFragment.handleResult(SearchInfo)  (org.schabi.newpipe.fragments.list.search)
            BaseListInfoFragment.handleResult(I)  (org.schabi.newpipe.fragments.list)
            SubscriptionFragment.handleResult(SubscriptionState)  (org.schabi.newpipe.local.subscription)
            VideoDetailFragment.prepareAndHandleInfo(StreamInfo, boolean)  (org.schabi.newpipe.fragments.detail)
            Anonymous in getPlaylistsSubscriber() in BookmarkFragment.onNext(List<PlaylistLocalItem>)  (org.schabi.newpipe.local.bookmark)
            BaseListInfoFragment.doInitialLoadLogic()  (org.schabi.newpipe.fragments.list)
            KioskFragment.handleResult(KioskInfo)  (org.schabi.newpipe.fragments.list.kiosk)
            PlaylistFragment.handleResult(PlaylistInfo)  (org.schabi.newpipe.fragments.list.playlist)
            FeedFragment.onResume()  (org.schabi.newpipe.local.feed)
            Anonymous in getHistoryObserver() in StatisticsPlaylistFragment.onNext(List<StreamStatisticsEntry>)  (org.schabi.newpipe.local.history)
            FeedFragment.onViewCreated(View, Bundle)  (org.schabi.newpipe.local.feed)
            BaseListInfoFragment.startLoading(boolean)  (org.schabi.newpipe.fragments.list)
            RelatedItemsFragment.handleResult(RelatedItemInfo)  (org.schabi.newpipe.fragments.list.videos)
            Anonymous in getPlaylistObserver() in LocalPlaylistFragment.onNext(List<PlaylistStreamEntry>)  (org.schabi.newpipe.local.playlist)
            VideoDetailFragment.runWorker(boolean, boolean)  (org.schabi.newpipe.fragments.detail)
            ChannelFragment.handleResult(ChannelInfo)  (org.schabi.newpipe.fragments.list.channel)
            CommentsFragment.handleResult(CommentsInfo)  (org.schabi.newpipe.fragments.list.comments)
            SearchFragment.startLoading(boolean)  (org.schabi.newpipe.fragments.list.search)
            SubscriptionFragment.initViews(View, Bundle)  (org.schabi.newpipe.local.subscription)
        ChannelFragment.handleResult(ChannelInfo)(2 usages)  (org.schabi.newpipe.fragments.list.channel)
            SubscriptionFragment.handleResult(SubscriptionState)  (org.schabi.newpipe.local.subscription)
            VideoDetailFragment.prepareAndHandleInfo(StreamInfo, boolean)  (org.schabi.newpipe.fragments.detail)
            BookmarkFragment.handleResult(List<PlaylistLocalItem>)  (org.schabi.newpipe.local.bookmark)
            StatisticsPlaylistFragment.handleResult(List<StreamStatisticsEntry>)  (org.schabi.newpipe.local.history)
            Anonymous in getPlaylistsSubscriber() in BookmarkFragment.onNext(List<PlaylistLocalItem>)  (org.schabi.newpipe.local.bookmark)
            BaseListInfoFragment.doInitialLoadLogic()  (org.schabi.newpipe.fragments.list)
            FeedFragment.onResume()  (org.schabi.newpipe.local.feed)
            Anonymous in getHistoryObserver() in StatisticsPlaylistFragment.onNext(List<StreamStatisticsEntry>)  (org.schabi.newpipe.local.history)
            FeedFragment.onViewCreated(View, Bundle)  (org.schabi.newpipe.local.feed)
            BaseListInfoFragment.startLoading(boolean)  (org.schabi.newpipe.fragments.list)
            VideoDetailFragment.runWorker(boolean, boolean)  (org.schabi.newpipe.fragments.detail)
            Anonymous in getPlaylistObserver() in LocalPlaylistFragment.onNext(List<PlaylistStreamEntry>)  (org.schabi.newpipe.local.playlist)
            LocalPlaylistFragment.handleResult(List<PlaylistStreamEntry>)  (org.schabi.newpipe.local.playlist)
            SearchFragment.startLoading(boolean)  (org.schabi.newpipe.fragments.list.search)
            SubscriptionFragment.initViews(View, Bundle)  (org.schabi.newpipe.local.subscription)
        VideoDetailFragment.openNormalBackgroundPlayer(boolean)  (org.schabi.newpipe.fragments.detail)
            VideoDetailFragment.openBackgroundPlayer(boolean)  (org.schabi.newpipe.fragments.detail)
        NavigationHelper.enqueueOnPlayer(Context, PlayQueue)  (org.schabi.newpipe.util)
            StreamDialogEntry.enqueue  (org.schabi.newpipe.util)
        VideoDetailFragment.openPopupPlayer(boolean)  (org.schabi.newpipe.fragments.detail)
            VideoDetailFragment.onLongClick(View)  (org.schabi.newpipe.fragments.detail)
            VideoDetailFragment.onClick(View)  (org.schabi.newpipe.fragments.detail)
        PlaylistFragment.handleResult(PlaylistInfo)(2 usages)  (org.schabi.newpipe.fragments.list.playlist)
            SubscriptionFragment.handleResult(SubscriptionState)  (org.schabi.newpipe.local.subscription)
            VideoDetailFragment.prepareAndHandleInfo(StreamInfo, boolean)  (org.schabi.newpipe.fragments.detail)
            BookmarkFragment.handleResult(List<PlaylistLocalItem>)  (org.schabi.newpipe.local.bookmark)
            StatisticsPlaylistFragment.handleResult(List<StreamStatisticsEntry>)  (org.schabi.newpipe.local.history)
            Anonymous in getPlaylistsSubscriber() in BookmarkFragment.onNext(List<PlaylistLocalItem>)  (org.schabi.newpipe.local.bookmark)
            BaseListInfoFragment.doInitialLoadLogic()  (org.schabi.newpipe.fragments.list)
            FeedFragment.onResume()  (org.schabi.newpipe.local.feed)
            FeedFragment.onViewCreated(View, Bundle)  (org.schabi.newpipe.local.feed)
            Anonymous in getHistoryObserver() in StatisticsPlaylistFragment.onNext(List<StreamStatisticsEntry>)  (org.schabi.newpipe.local.history)
            BaseListInfoFragment.startLoading(boolean)  (org.schabi.newpipe.fragments.list)
            Anonymous in getPlaylistObserver() in LocalPlaylistFragment.onNext(List<PlaylistStreamEntry>)  (org.schabi.newpipe.local.playlist)
            VideoDetailFragment.runWorker(boolean, boolean)  (org.schabi.newpipe.fragments.detail)
            LocalPlaylistFragment.handleResult(List<PlaylistStreamEntry>)  (org.schabi.newpipe.local.playlist)
            SearchFragment.startLoading(boolean)  (org.schabi.newpipe.fragments.list.search)
            SubscriptionFragment.initViews(View, Bundle)  (org.schabi.newpipe.local.subscription)
NavigationHelper.playOnBackgroundPlayer(Context, PlayQueue, boolean)  (org.schabi.newpipe.util)
    PlaylistFragment.showStreamDialog(StreamInfoItem)  (org.schabi.newpipe.fragments.list.playlist)
        Anonymous in initListeners() in BaseListFragment.held(StreamInfoItem)  (org.schabi.newpipe.fragments.list)
    StatisticsPlaylistFragment.showStreamDialog(StreamStatisticsEntry)  (org.schabi.newpipe.local.history)
        Anonymous in initListeners() in StatisticsPlaylistFragment.held(LocalItem)  (org.schabi.newpipe.local.history)
    FetcherService in RouterActivity.getResultHandler(Choice)  (org.schabi.newpipe)
        FetcherService in RouterActivity.handleChoice(Choice)  (org.schabi.newpipe)
    LocalPlaylistFragment.handleResult(List<PlaylistStreamEntry>)  (org.schabi.newpipe.local.playlist)
        SearchFragment.handleResult(SearchInfo)  (org.schabi.newpipe.fragments.list.search)
        BaseListInfoFragment.handleResult(I)  (org.schabi.newpipe.fragments.list)
        SubscriptionFragment.handleResult(SubscriptionState)  (org.schabi.newpipe.local.subscription)
        VideoDetailFragment.prepareAndHandleInfo(StreamInfo, boolean)  (org.schabi.newpipe.fragments.detail)
        Anonymous in getPlaylistsSubscriber() in BookmarkFragment.onNext(List<PlaylistLocalItem>)  (org.schabi.newpipe.local.bookmark)
        BaseListInfoFragment.doInitialLoadLogic()  (org.schabi.newpipe.fragments.list)
        PlaylistFragment.handleResult(PlaylistInfo)  (org.schabi.newpipe.fragments.list.playlist)
        KioskFragment.handleResult(KioskInfo)  (org.schabi.newpipe.fragments.list.kiosk)
        FeedFragment.onResume()  (org.schabi.newpipe.local.feed)
        FeedFragment.onViewCreated(View, Bundle)  (org.schabi.newpipe.local.feed)
        Anonymous in getHistoryObserver() in StatisticsPlaylistFragment.onNext(List<StreamStatisticsEntry>)  (org.schabi.newpipe.local.history)
        BaseListInfoFragment.startLoading(boolean)  (org.schabi.newpipe.fragments.list)
        RelatedItemsFragment.handleResult(RelatedItemInfo)  (org.schabi.newpipe.fragments.list.videos)
        Anonymous in getPlaylistObserver() in LocalPlaylistFragment.onNext(List<PlaylistStreamEntry>)  (org.schabi.newpipe.local.playlist)
        VideoDetailFragment.runWorker(boolean, boolean)  (org.schabi.newpipe.fragments.detail)
        CommentsFragment.handleResult(CommentsInfo)  (org.schabi.newpipe.fragments.list.comments)
        ChannelFragment.handleResult(ChannelInfo)  (org.schabi.newpipe.fragments.list.channel)
        SearchFragment.startLoading(boolean)  (org.schabi.newpipe.fragments.list.search)
        SubscriptionFragment.initViews(View, Bundle)  (org.schabi.newpipe.local.subscription)
    StreamDialogEntry.start_here_on_background  (org.schabi.newpipe.util)
        BaseListFragment.showStreamDialog(StreamInfoItem)  (org.schabi.newpipe.fragments.list)
        BaseListFragment.showStreamDialog(StreamInfoItem)  (org.schabi.newpipe.fragments.list)
        StatisticsPlaylistFragment.showStreamDialog(StreamStatisticsEntry)  (org.schabi.newpipe.local.history)
        LocalPlaylistFragment.showStreamItemDialog(PlaylistStreamEntry)  (org.schabi.newpipe.local.playlist)
        LocalPlaylistFragment.showStreamItemDialog(PlaylistStreamEntry)  (org.schabi.newpipe.local.playlist)
        StatisticsPlaylistFragment.showStreamDialog(StreamStatisticsEntry)  (org.schabi.newpipe.local.history)
        StatisticsPlaylistFragment.showStreamDialog(StreamStatisticsEntry)  (org.schabi.newpipe.local.history)
        LocalPlaylistFragment.showStreamItemDialog(PlaylistStreamEntry)  (org.schabi.newpipe.local.playlist)
        [Invalid]
        PlaylistFragment.showStreamDialog(StreamInfoItem)  (org.schabi.newpipe.fragments.list.playlist)
        PlaylistFragment.showStreamDialog(StreamInfoItem)  (org.schabi.newpipe.fragments.list.playlist)
        [Invalid]
        PlaylistFragment.showStreamDialog(StreamInfoItem)  (org.schabi.newpipe.fragments.list.playlist)
    StatisticsPlaylistFragment.handleResult(List<StreamStatisticsEntry>)  (org.schabi.newpipe.local.history)
        SearchFragment.handleResult(SearchInfo)  (org.schabi.newpipe.fragments.list.search)
        BaseListInfoFragment.handleResult(I)  (org.schabi.newpipe.fragments.list)
        SubscriptionFragment.handleResult(SubscriptionState)  (org.schabi.newpipe.local.subscription)
        VideoDetailFragment.prepareAndHandleInfo(StreamInfo, boolean)  (org.schabi.newpipe.fragments.detail)
        Anonymous in getPlaylistsSubscriber() in BookmarkFragment.onNext(List<PlaylistLocalItem>)  (org.schabi.newpipe.local.bookmark)
        BaseListInfoFragment.doInitialLoadLogic()  (org.schabi.newpipe.fragments.list)
        PlaylistFragment.handleResult(PlaylistInfo)  (org.schabi.newpipe.fragments.list.playlist)
        KioskFragment.handleResult(KioskInfo)  (org.schabi.newpipe.fragments.list.kiosk)
        FeedFragment.onResume()  (org.schabi.newpipe.local.feed)
        Anonymous in getHistoryObserver() in StatisticsPlaylistFragment.onNext(List<StreamStatisticsEntry>)  (org.schabi.newpipe.local.history)
        FeedFragment.onViewCreated(View, Bundle)  (org.schabi.newpipe.local.feed)
        BaseListInfoFragment.startLoading(boolean)  (org.schabi.newpipe.fragments.list)
        RelatedItemsFragment.handleResult(RelatedItemInfo)  (org.schabi.newpipe.fragments.list.videos)
        Anonymous in getPlaylistObserver() in LocalPlaylistFragment.onNext(List<PlaylistStreamEntry>)  (org.schabi.newpipe.local.playlist)
        VideoDetailFragment.runWorker(boolean, boolean)  (org.schabi.newpipe.fragments.detail)
        ChannelFragment.handleResult(ChannelInfo)  (org.schabi.newpipe.fragments.list.channel)
        CommentsFragment.handleResult(CommentsInfo)  (org.schabi.newpipe.fragments.list.comments)
        SearchFragment.startLoading(boolean)  (org.schabi.newpipe.fragments.list.search)
        SubscriptionFragment.initViews(View, Bundle)  (org.schabi.newpipe.local.subscription)
    ChannelFragment.handleResult(ChannelInfo)  (org.schabi.newpipe.fragments.list.channel)
        SubscriptionFragment.handleResult(SubscriptionState)  (org.schabi.newpipe.local.subscription)
        BookmarkFragment.handleResult(List<PlaylistLocalItem>)  (org.schabi.newpipe.local.bookmark)
        VideoDetailFragment.prepareAndHandleInfo(StreamInfo, boolean)  (org.schabi.newpipe.fragments.detail)
        StatisticsPlaylistFragment.handleResult(List<StreamStatisticsEntry>)  (org.schabi.newpipe.local.history)
        Anonymous in getPlaylistsSubscriber() in BookmarkFragment.onNext(List<PlaylistLocalItem>)  (org.schabi.newpipe.local.bookmark)
        BaseListInfoFragment.doInitialLoadLogic()  (org.schabi.newpipe.fragments.list)
        FeedFragment.onResume()  (org.schabi.newpipe.local.feed)
        Anonymous in getHistoryObserver() in StatisticsPlaylistFragment.onNext(List<StreamStatisticsEntry>)  (org.schabi.newpipe.local.history)
        FeedFragment.onViewCreated(View, Bundle)  (org.schabi.newpipe.local.feed)
        BaseListInfoFragment.startLoading(boolean)  (org.schabi.newpipe.fragments.list)
        Anonymous in getPlaylistObserver() in LocalPlaylistFragment.onNext(List<PlaylistStreamEntry>)  (org.schabi.newpipe.local.playlist)
        VideoDetailFragment.runWorker(boolean, boolean)  (org.schabi.newpipe.fragments.detail)
        LocalPlaylistFragment.handleResult(List<PlaylistStreamEntry>)  (org.schabi.newpipe.local.playlist)
        SearchFragment.startLoading(boolean)  (org.schabi.newpipe.fragments.list.search)
        SubscriptionFragment.initViews(View, Bundle)  (org.schabi.newpipe.local.subscription)
    LocalPlaylistFragment.showStreamItemDialog(PlaylistStreamEntry)  (org.schabi.newpipe.local.playlist)
        Anonymous in initListeners() in LocalPlaylistFragment.held(LocalItem)  (org.schabi.newpipe.local.playlist)
    PlayQueueActivity.onOptionsItemSelected(MenuItem)  (org.schabi.newpipe.player)
        PreferenceActivity.onOptionsItemSelected(MenuItem)  (android.preference)
        Activity.onMenuItemSelected(int, MenuItem)  (android.app)
        Activity.onOptionsItemSelected(MenuItem)  (android.app)
        NavigatingActivity.onOptionsItemSelected(MenuItem)  (leakcanary.internal.navigation)
        AboutActivity.onOptionsItemSelected(MenuItem)  (org.schabi.newpipe.about)
    VideoDetailFragment.openNormalBackgroundPlayer(boolean)  (org.schabi.newpipe.fragments.detail)
        VideoDetailFragment.openBackgroundPlayer(boolean)  (org.schabi.newpipe.fragments.detail)
    PlaylistFragment.handleResult(PlaylistInfo)  (org.schabi.newpipe.fragments.list.playlist)
        SubscriptionFragment.handleResult(SubscriptionState)  (org.schabi.newpipe.local.subscription)
        BookmarkFragment.handleResult(List<PlaylistLocalItem>)  (org.schabi.newpipe.local.bookmark)
        VideoDetailFragment.prepareAndHandleInfo(StreamInfo, boolean)  (org.schabi.newpipe.fragments.detail)
        StatisticsPlaylistFragment.handleResult(List<StreamStatisticsEntry>)  (org.schabi.newpipe.local.history)
        Anonymous in getPlaylistsSubscriber() in BookmarkFragment.onNext(List<PlaylistLocalItem>)  (org.schabi.newpipe.local.bookmark)
        BaseListInfoFragment.doInitialLoadLogic()  (org.schabi.newpipe.fragments.list)
        FeedFragment.onResume()  (org.schabi.newpipe.local.feed)
        Anonymous in getHistoryObserver() in StatisticsPlaylistFragment.onNext(List<StreamStatisticsEntry>)  (org.schabi.newpipe.local.history)
        FeedFragment.onViewCreated(View, Bundle)  (org.schabi.newpipe.local.feed)
        BaseListInfoFragment.startLoading(boolean)  (org.schabi.newpipe.fragments.list)
        Anonymous in getPlaylistObserver() in LocalPlaylistFragment.onNext(List<PlaylistStreamEntry>)  (org.schabi.newpipe.local.playlist)
        VideoDetailFragment.runWorker(boolean, boolean)  (org.schabi.newpipe.fragments.detail)
        LocalPlaylistFragment.handleResult(List<PlaylistStreamEntry>)  (org.schabi.newpipe.local.playlist)
        SearchFragment.startLoading(boolean)  (org.schabi.newpipe.fragments.list.search)
        SubscriptionFragment.initViews(View, Bundle)  (org.schabi.newpipe.local.subscription)
NotificationUtil.getIntentForNotification(Player)  (org.schabi.newpipe.player)
    NotificationUtil.updateNotification(Player)  (org.schabi.newpipe.player)
        NotificationUtil.createNotificationIfNeededAndUpdate(Player, boolean)  (org.schabi.newpipe.player)
            Player.onPlayQueueEdited()  (org.schabi.newpipe.player)
            Player.onShuffleOrRepeatModeChanged()  (org.schabi.newpipe.player)
            Player.onMetadataChanged(MediaSourceTag)  (org.schabi.newpipe.player)
            Player.onPlaying()  (org.schabi.newpipe.player)
            Player.onBroadcastReceived(Intent)  (org.schabi.newpipe.player)
            Anonymous in initThumbnail() in Player.onBitmapFailed(Exception, Drawable)  (org.schabi.newpipe.player)
            Player.onBuffering()  (org.schabi.newpipe.player)
            Player.onTimelineChanged(Timeline, int)  (org.schabi.newpipe.player)
            Player.onBlocked()  (org.schabi.newpipe.player)
            Anonymous in initThumbnail() in Player.onBitmapLoaded(Bitmap, LoadedFrom)  (org.schabi.newpipe.player)
            Player.onPaused()  (org.schabi.newpipe.player)
            Player.onPausedSeek()  (org.schabi.newpipe.player)
            Player.onCompleted()  (org.schabi.newpipe.player)
        NotificationUtil.createNotificationAndStartForeground(Player, Service)  (org.schabi.newpipe.player)
            MainPlayer.createView()  (org.schabi.newpipe.player)
            MainPlayer.onStartCommand(Intent, int, int)  (org.schabi.newpipe.player)
NavigationHelper.getPlayerEnqueueNextIntent(Context, Class<T>, PlayQueue)  (org.schabi.newpipe.util)
    NavigationHelper.enqueueNextOnPlayer(Context, PlayQueue)  (org.schabi.newpipe.util)
        StreamDialogEntry.enqueue_next  (org.schabi.newpipe.util)
            PlaylistFragment.showStreamDialog(StreamInfoItem)  (org.schabi.newpipe.fragments.list.playlist)
                Anonymous in initListeners() in BaseListFragment.held(StreamInfoItem)  (org.schabi.newpipe.fragments.list)
            LocalPlaylistFragment.showStreamItemDialog(PlaylistStreamEntry)  (org.schabi.newpipe.local.playlist)
                Anonymous in initListeners() in LocalPlaylistFragment.held(LocalItem)  (org.schabi.newpipe.local.playlist)
            StatisticsPlaylistFragment.showStreamDialog(StreamStatisticsEntry)  (org.schabi.newpipe.local.history)
                Anonymous in initListeners() in StatisticsPlaylistFragment.held(LocalItem)  (org.schabi.newpipe.local.history)
            BaseListFragment.showStreamDialog(StreamInfoItem)  (org.schabi.newpipe.fragments.list)
                Anonymous in initListeners() in BaseListFragment.held(StreamInfoItem)  (org.schabi.newpipe.fragments.list)
            [Invalid]
NavigationHelper.playOnPopupPlayer(Context, PlayQueue, boolean)  (org.schabi.newpipe.util)
    Player.onFragmentStopped()  (org.schabi.newpipe.player)
        Player.onBroadcastReceived(Intent)(2 usages)  (org.schabi.newpipe.player)
    FetcherService in RouterActivity.getResultHandler(Choice)  (org.schabi.newpipe)
        FetcherService in RouterActivity.handleChoice(Choice)  (org.schabi.newpipe)
    LocalPlaylistFragment.handleResult(List<PlaylistStreamEntry>)  (org.schabi.newpipe.local.playlist)
        SearchFragment.handleResult(SearchInfo)  (org.schabi.newpipe.fragments.list.search)
        BaseListInfoFragment.handleResult(I)  (org.schabi.newpipe.fragments.list)
        SubscriptionFragment.handleResult(SubscriptionState)  (org.schabi.newpipe.local.subscription)
        VideoDetailFragment.prepareAndHandleInfo(StreamInfo, boolean)  (org.schabi.newpipe.fragments.detail)
        Anonymous in getPlaylistsSubscriber() in BookmarkFragment.onNext(List<PlaylistLocalItem>)  (org.schabi.newpipe.local.bookmark)
        BaseListInfoFragment.doInitialLoadLogic()  (org.schabi.newpipe.fragments.list)
        KioskFragment.handleResult(KioskInfo)  (org.schabi.newpipe.fragments.list.kiosk)
        PlaylistFragment.handleResult(PlaylistInfo)  (org.schabi.newpipe.fragments.list.playlist)
        FeedFragment.onResume()  (org.schabi.newpipe.local.feed)
        FeedFragment.onViewCreated(View, Bundle)  (org.schabi.newpipe.local.feed)
        Anonymous in getHistoryObserver() in StatisticsPlaylistFragment.onNext(List<StreamStatisticsEntry>)  (org.schabi.newpipe.local.history)
        BaseListInfoFragment.startLoading(boolean)  (org.schabi.newpipe.fragments.list)
        RelatedItemsFragment.handleResult(RelatedItemInfo)  (org.schabi.newpipe.fragments.list.videos)
        Anonymous in getPlaylistObserver() in LocalPlaylistFragment.onNext(List<PlaylistStreamEntry>)  (org.schabi.newpipe.local.playlist)
        VideoDetailFragment.runWorker(boolean, boolean)  (org.schabi.newpipe.fragments.detail)
        ChannelFragment.handleResult(ChannelInfo)  (org.schabi.newpipe.fragments.list.channel)
        CommentsFragment.handleResult(CommentsInfo)  (org.schabi.newpipe.fragments.list.comments)
        SearchFragment.startLoading(boolean)  (org.schabi.newpipe.fragments.list.search)
        SubscriptionFragment.initViews(View, Bundle)  (org.schabi.newpipe.local.subscription)
    ChannelFragment.handleResult(ChannelInfo)  (org.schabi.newpipe.fragments.list.channel)
        SubscriptionFragment.handleResult(SubscriptionState)  (org.schabi.newpipe.local.subscription)
        VideoDetailFragment.prepareAndHandleInfo(StreamInfo, boolean)  (org.schabi.newpipe.fragments.detail)
        BookmarkFragment.handleResult(List<PlaylistLocalItem>)  (org.schabi.newpipe.local.bookmark)
        StatisticsPlaylistFragment.handleResult(List<StreamStatisticsEntry>)  (org.schabi.newpipe.local.history)
        Anonymous in getPlaylistsSubscriber() in BookmarkFragment.onNext(List<PlaylistLocalItem>)  (org.schabi.newpipe.local.bookmark)
        BaseListInfoFragment.doInitialLoadLogic()  (org.schabi.newpipe.fragments.list)
        FeedFragment.onResume()  (org.schabi.newpipe.local.feed)
        Anonymous in getHistoryObserver() in StatisticsPlaylistFragment.onNext(List<StreamStatisticsEntry>)  (org.schabi.newpipe.local.history)
        FeedFragment.onViewCreated(View, Bundle)  (org.schabi.newpipe.local.feed)
        BaseListInfoFragment.startLoading(boolean)  (org.schabi.newpipe.fragments.list)
        Anonymous in getPlaylistObserver() in LocalPlaylistFragment.onNext(List<PlaylistStreamEntry>)  (org.schabi.newpipe.local.playlist)
        VideoDetailFragment.runWorker(boolean, boolean)  (org.schabi.newpipe.fragments.detail)
        LocalPlaylistFragment.handleResult(List<PlaylistStreamEntry>)  (org.schabi.newpipe.local.playlist)
        SearchFragment.startLoading(boolean)  (org.schabi.newpipe.fragments.list.search)
        SubscriptionFragment.initViews(View, Bundle)  (org.schabi.newpipe.local.subscription)
    StatisticsPlaylistFragment.handleResult(List<StreamStatisticsEntry>)  (org.schabi.newpipe.local.history)
        SearchFragment.handleResult(SearchInfo)  (org.schabi.newpipe.fragments.list.search)
        BaseListInfoFragment.handleResult(I)  (org.schabi.newpipe.fragments.list)
        SubscriptionFragment.handleResult(SubscriptionState)  (org.schabi.newpipe.local.subscription)
        VideoDetailFragment.prepareAndHandleInfo(StreamInfo, boolean)  (org.schabi.newpipe.fragments.detail)
        Anonymous in getPlaylistsSubscriber() in BookmarkFragment.onNext(List<PlaylistLocalItem>)  (org.schabi.newpipe.local.bookmark)
        BaseListInfoFragment.doInitialLoadLogic()  (org.schabi.newpipe.fragments.list)
        PlaylistFragment.handleResult(PlaylistInfo)  (org.schabi.newpipe.fragments.list.playlist)
        KioskFragment.handleResult(KioskInfo)  (org.schabi.newpipe.fragments.list.kiosk)
        FeedFragment.onResume()  (org.schabi.newpipe.local.feed)
        Anonymous in getHistoryObserver() in StatisticsPlaylistFragment.onNext(List<StreamStatisticsEntry>)  (org.schabi.newpipe.local.history)
        FeedFragment.onViewCreated(View, Bundle)  (org.schabi.newpipe.local.feed)
        BaseListInfoFragment.startLoading(boolean)  (org.schabi.newpipe.fragments.list)
        RelatedItemsFragment.handleResult(RelatedItemInfo)  (org.schabi.newpipe.fragments.list.videos)
        Anonymous in getPlaylistObserver() in LocalPlaylistFragment.onNext(List<PlaylistStreamEntry>)  (org.schabi.newpipe.local.playlist)
        VideoDetailFragment.runWorker(boolean, boolean)  (org.schabi.newpipe.fragments.detail)
        ChannelFragment.handleResult(ChannelInfo)  (org.schabi.newpipe.fragments.list.channel)
        CommentsFragment.handleResult(CommentsInfo)  (org.schabi.newpipe.fragments.list.comments)
        SearchFragment.startLoading(boolean)  (org.schabi.newpipe.fragments.list.search)
        SubscriptionFragment.initViews(View, Bundle)  (org.schabi.newpipe.local.subscription)
    StreamDialogEntry.start_here_on_popup  (org.schabi.newpipe.util)
        BaseListFragment.showStreamDialog(StreamInfoItem)  (org.schabi.newpipe.fragments.list)
        StatisticsPlaylistFragment.showStreamDialog(StreamStatisticsEntry)  (org.schabi.newpipe.local.history)
        LocalPlaylistFragment.showStreamItemDialog(PlaylistStreamEntry)  (org.schabi.newpipe.local.playlist)
        PlaylistFragment.showStreamDialog(StreamInfoItem)  (org.schabi.newpipe.fragments.list.playlist)
        [Invalid]
    PlayQueueActivity.onOptionsItemSelected(MenuItem)  (org.schabi.newpipe.player)
        PreferenceActivity.onOptionsItemSelected(MenuItem)  (android.preference)
        Activity.onMenuItemSelected(int, MenuItem)  (android.app)
        Activity.onOptionsItemSelected(MenuItem)  (android.app)
        NavigatingActivity.onOptionsItemSelected(MenuItem)  (leakcanary.internal.navigation)
        AboutActivity.onOptionsItemSelected(MenuItem)  (org.schabi.newpipe.about)
    VideoDetailFragment.openPopupPlayer(boolean)  (org.schabi.newpipe.fragments.detail)
        VideoDetailFragment.onLongClick(View)  (org.schabi.newpipe.fragments.detail)
        VideoDetailFragment.onClick(View)  (org.schabi.newpipe.fragments.detail)
    PlaylistFragment.handleResult(PlaylistInfo)  (org.schabi.newpipe.fragments.list.playlist)
        SubscriptionFragment.handleResult(SubscriptionState)  (org.schabi.newpipe.local.subscription)
        VideoDetailFragment.prepareAndHandleInfo(StreamInfo, boolean)  (org.schabi.newpipe.fragments.detail)
        BookmarkFragment.handleResult(List<PlaylistLocalItem>)  (org.schabi.newpipe.local.bookmark)
        StatisticsPlaylistFragment.handleResult(List<StreamStatisticsEntry>)  (org.schabi.newpipe.local.history)
        Anonymous in getPlaylistsSubscriber() in BookmarkFragment.onNext(List<PlaylistLocalItem>)  (org.schabi.newpipe.local.bookmark)
        BaseListInfoFragment.doInitialLoadLogic()  (org.schabi.newpipe.fragments.list)
        FeedFragment.onResume()  (org.schabi.newpipe.local.feed)
        Anonymous in getHistoryObserver() in StatisticsPlaylistFragment.onNext(List<StreamStatisticsEntry>)  (org.schabi.newpipe.local.history)
        FeedFragment.onViewCreated(View, Bundle)  (org.schabi.newpipe.local.feed)
        BaseListInfoFragment.startLoading(boolean)  (org.schabi.newpipe.fragments.list)
        Anonymous in getPlaylistObserver() in LocalPlaylistFragment.onNext(List<PlaylistStreamEntry>)  (org.schabi.newpipe.local.playlist)
        VideoDetailFragment.runWorker(boolean, boolean)  (org.schabi.newpipe.fragments.detail)
        LocalPlaylistFragment.handleResult(List<PlaylistStreamEntry>)  (org.schabi.newpipe.local.playlist)
        SearchFragment.startLoading(boolean)  (org.schabi.newpipe.fragments.list.search)
        SubscriptionFragment.initViews(View, Bundle)  (org.schabi.newpipe.local.subscription)
    InternalUrlsHandler.playOnPopup(Context, String, StreamingService, int, CompositeDisposable)  (org.schabi.newpipe.util.external_communication)
        Anonymous in addClickListenersOnTimestamps() in TextLinkifier.onClick(View)  (org.schabi.newpipe.util.external_communication)
        InternalUrlsHandler.handleUrl(Context, String, Pattern, CompositeDisposable)  (org.schabi.newpipe.util.external_communication)
```

</details>

Did a quick and dirty fix with #7668 instead. Someone should do a followup PR where the code is refactored.

**Comment by ildar** [comment]
> a spaghetti monster

No wonder I experience race condition crashes for years.

**Comment by werjany** [comment]
Same problem still appears on Android 12, NewPipe v0.22.1 (Xiaomi, MIUI 13)
And moreover a video jumps to beginning also when locking screen, not only when switching between playing within the app and using popup.
